### PR TITLE
Dialyzer and better travis

### PR DESCRIPTION
**Pull request by binarin**
- Make `dialyzer` check successful
- Get rid of `rebar3` for tests - it's possible to run tests with coverage using rabbit-provided makefiles
- Switched back to travis-provided erlang

### CODE REVIEW

File: rabbitmq-components.mk [hunk 1]
```diff
@@ -44,6 +44,7 @@ dep_rabbitmq_event_exchange           = git_rmq rabbitmq-event-exchange $(curren
 dep_rabbitmq_federation               = git_rmq rabbitmq-federation $(current_rmq_ref) $(base_rmq_ref) master
 dep_rabbitmq_federation_management    = git_rmq rabbitmq-federation-management $(current_rmq_ref) $(base_rmq_ref) master
 dep_rabbitmq_java_client              = git_rmq rabbitmq-java-client $(current_rmq_ref) $(base_rmq_ref) master
+dep_rabbitmq_jms_client               = git_rmq rabbitmq-jms-client $(current_rmq_ref) $(base_rmq_ref) master
```
Comment (gmr) [inline rabbitmq-components.mk]: One of the reasons I didn't want to use erlang.mk was the requirement I have java installed to work. Another is how long tests take to run. I imagine this makes that situation worse?

File: src/autocluster.erl [hunk 2]
```diff
@@ -376,9 +371,7 @@ maybe_delay_startup() ->
 -spec startup_delay(integer()) -> ok.
 startup_delay(0) -> ok;
 startup_delay(Max) ->
-  <<A:32, B:32, C:32>> = crypto:rand_bytes(12),
-  random:seed({A,B,C}),
-  Duration = random:uniform(Max),
+  Duration = rand_compat:uniform(Max),
```
Comment (gmr) [inline src/autocluster.erl]: Why the change here? What does this do for additional dependency plugins?

File: rabbitmq-components.mk (hunk 1 above)
Comment (binarin) [inline rabbitmq-components.mk]: This exact line is not about java, but just about broken build - like in https://github.com/aweber/rabbitmq-autocluster/pull/77

And I've added a commit to this PR that removes java dependency completely.

Main speed improvement of `rebar3` is that in never rebuilds dependencies, which I'm not sure is a good thing when working on rabbit: I frequently change code in more than one project, i.e. `rabbit_common` and `rabbit`.

Also I'm going to introduce some integration tests, and with new shiny Common Test support in rabbit it's more easy than before (with all that `WITH_BROKER_TEST_COMMANDS` and etc.).  So it's either completely switching to common test thing, or maintaining 2 different test frameworks.

File: src/autocluster.erl (hunk 2 above)
Comment (binarin) [inline src/autocluster.erl]: `random` is deprecated in erlang 19, and `rand_compat` (which is is `rabbit_common`) already handles all things related to that deprecation - i.e. whether `rand` or `random` is available.

**Comment by michaelklishin** [comment]
Keep in mind that `rand_compat` is very very new in `rabbit_common`.

**Comment by binarin** [comment]
@michaelklishin Yes, replaced it with a call to `rabbit_misc:random/1`

**Comment by binarin** [comment]
BTW, `make ct SKIP_DEPS=true` runs tests way more faster during development.
